### PR TITLE
typescript: Add TypeScript migration notice to README file

### PR DESCRIPTION
**Pull request by pdcastro**
Closes: #733 
Change-type: patch
Signed-off-by: Paulo Castro <paulo@balena.io>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -123,13 +123,20 @@ You can also get in touch with us in the balena [forums](https://forums.balena.i
...
+relatively small and self-contained to help with iterations of code review.
+
 After cloning this repository and running `npm install` you can build the CLI using `npm run build`.
 You can then run the generated build using `./bin/balena`.
 In order to ease development:
 * you can build the CLI using the `npm run build:fast` variant which skips some of the build steps or
 * you can use `./bin/balena-dev` which live transpiles the sources of the CLI.
 
-In either case, before opening a PR make sure to also test your changes after doing a full build with `npm run build`.
+Before opening a pull request, please make sure to test your changes with `npm test`.
```
[the remark this answers is not in the document]
Comment (pdcastro) [reply README.md]: I found that `npm test` already automatically calls `npm run build`:
```
    "pretest": "npm run build",
    "test": "gulp test",
```
But I have undone this change in order to narrow the scope of the pull request.

File: README.md [hunk 2]
```diff
@@ -123,13 +123,20 @@ You can also get in touch with us in the balena [forums](https://forums.balena.i
 Development guidelines
 ----------------------
 
+Programming language: the CLI was originally written in [CoffeeScript](https://coffeescript.org),
+but we have decided to migrate to [TypeScript](https://www.typescriptlang.org/) in order to take
+advantage of static typing and formal programming interfaces. The migration is taking place
+gradually, as part of maintenance work or new features. We also welcome pull requests whose sole
+purpose are to migrate from CoffeeScript to TypeScript; in this case, please keep pull requests
+relatively small and self-contained to help with iterations of code review.
```
[the remark this answers is not in the document]
Comment (pdcastro) [reply README.md]: Yeah, I could think of improvements to almost every single paragraph of the README file, but the only reason I'm submitting this PR is to close issue #733 (I've started looking at several old issues as part of the 2019 roadmap preparation). I think the README file could do with an overall revision under a separate issue and PR.

[the remark this answers is not in the document]
Comment (thgreasi) [reply README.md]: Oh that makes sense :

File: README.md [hunk 3]
```diff
@@ -123,6 +123,11 @@ You can also get in touch with us in the balena [forums](https://forums.balena.i
 Development guidelines
 ----------------------
 
+Programming language: the CLI was originally written in [CoffeeScript](https://coffeescript.org),
```
Comment (thgreasi) [inline README.md]: Could probably even drop the `Programming language:` part
```suggestion
The CLI was originally written in [CoffeeScript](https://coffeescript.org),
```

**Pull request merged**